### PR TITLE
Fixes for v7.3.0 - Issue#1597

### LINKED ISSUE #1597: Multiple issues in the 0.7.3 release
I'm using this ticket to collect issues that I found in the 0.7.3 release.

**1. Tails file upload fails when a credDef is created and multi ledger support is enabled.**

See discussion on the chat: https://chat.hyperledger.org/channel/aries-cloudagent-python?msg=XodRcp7dFxmncAWma I still do not get this to work.

**2. Revocation notification has a wrong revocation id for V1 exchanges (V2 works as expected)**

For example the credential acked event has the following link to the revocation registry:

```json
{
        "rev_reg_id": "7D2jFxSPJP4zTAP9WAcJWo:4:7D2jFxSPJP4zTAP9WAcJWo:3:CL:571:CustomerBankRevoc:CL_ACCUM:01f6b385-28bc-420a-bcc7-00ac282b1d3c",
        "cred_rev_id": "5"
}
```
But in the revocation event the value of the revocation id becomes 6 instead of 5. 

```json
{
    "thread_id": "indy::7D2jFxSPJP4zTAP9WAcJWo:4:7D2jFxSPJP4zTAP9WAcJWo:3:CL:571:CustomerBankRevoc:CL_ACCUM:01f6b385-28bc-420a-bcc7-00ac282b1d3c::6",
    "comment": null
}
```

**3a. V2 Credential exchange ignores the _auto-respond-credential-request: false_ flag**

If aca-py receives a V2 credential request it immediately responds even if the auto flag is set to false. Works as expected for V1.

**3b. V2 Presentation exchange ignores the _auto-respond-presentation-proposal: true_ flag**

If aca-py receives a V2 presentation proposal and the auto respond flag is set to true aca-py does not automatically respond. Works a expected for V1.

**4. Wallet type askar has issues when receiving V1 credentials.**

Probably same as or related to #1588 Any other time answering a credential offer with a request fails with:

```
2022-01-13 11:40:34,184 aries_cloudagent.admin.server ERROR Handler error with exception: Record not found.
2022-01-13 11:40:34,184 aries_askar.native.aries_askar.ffi.store INFO 	src/ffi/store.rs:1007 | Closed session SessionHandle(980)
2022-01-13 11:40:34,185 aries_askar.native.aries_askar.backend.db_utils INFO 	src/backend/db_utils.rs:168 | Dropped pool connection
Traceback (most recent call last):
  File "/home/indy/.pyenv/versions/3.6.13/lib/python3.6/site-packages/aries_cloudagent/protocols/issue_credential/v1_0/routes.py", line 997, in credential_exchange_send_request
    cred_ex_record, connection_record.my_did
  File "/home/indy/.pyenv/versions/3.6.13/lib/python3.6/site-packages/aries_cloudagent/protocols/issue_credential/v1_0/manager.py", line 471, in create_request
    await cred_ex_record.save(session, reason="create credential request")
  File "/home/indy/.pyenv/versions/3.6.13/lib/python3.6/site-packages/aries_cloudagent/messaging/models/base_record.py", line 354, in save
    await storage.update_record(record, record.value, record.tags)
  File "/home/indy/.pyenv/versions/3.6.13/lib/python3.6/site-packages/aries_cloudagent/storage/askar.py", line 121, in update_record
    raise StorageNotFoundError("Record not found") from None
aries_cloudagent.storage.error.StorageNotFoundError: Record not found

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/indy/.pyenv/versions/3.6.13/lib/python3.6/site-packages/aries_cloudagent/admin/server.py", line 163, in ready_middleware
    return await handler(request)
  File "/home/indy/.pyenv/versions/3.6.13/lib/python3.6/site-packages/aries_cloudagent/admin/server.py", line 200, in debug_middleware
    return await handler(request)
  File "/home/indy/.pyenv/versions/3.6.13/lib/python3.6/site-packages/aiohttp_apispec/middlewares.py", line 45, in validation_middleware
    return await handler(request)
  File "/home/indy/.pyenv/versions/3.6.13/lib/python3.6/site-packages/aries_cloudagent/admin/server.py", line 301, in check_token
    return await handler(request)
  File "/home/indy/.pyenv/versions/3.6.13/lib/python3.6/site-packages/aries_cloudagent/admin/server.py", line 391, in setup_context
    return await task
  File "/home/indy/.pyenv/versions/3.6.13/lib/python3.6/site-packages/aries_cloudagent/protocols/issue_credential/v1_0/routes.py", line 1017, in credential_exchange_send_request
    outbound_handler,
  File "/home/indy/.pyenv/versions/3.6.13/lib/python3.6/site-packages/aries_cloudagent/protocols/issue_credential/v1_0/__init__.py", line 63, in report_problem
    raise http_error_class(reason=err.roll_up) from err
aiohttp.web_exceptions.HTTPBadRequest: Record not found.
```

### PR DESCRIPTION
**Pull request by shaangill025**
Signed-off-by: Shaanjot Gill <shaangill025@users.noreply.github.com>
- resolve #1597 
- Fixes (1)

### CODE REVIEW

**Comment by shaangill025** [comment]
Regarding (3a), I believe it is working as intended:
Started Agent Alice
PORTS="5002 8002" ./scripts/run_docker start --admin 0.0.0.0 5002 --inbound-transport http 0.0.0.0 8002 --outbound-transport http -e http://host.docker.internal:8002 --admin-insecure-mode --public-invites --auto-ping-connection --auto-accept-invites --auto-accept-requests --auto-respond-credential-offer --auto-verify-presentation --genesis-url http://host.docker.internal:9000/genesis --seed 00000000000000000000000000000008 --auto-provision --recreate-wallet --wallet-type indy --wallet-name issuer --wallet-key mykey

Started Agent Bob
PORTS="5003 8003" ./scripts/run_docker start --admin 0.0.0.0 5003 --inbound-transport http 0.0.0.0 8003 --outbound-transport http -e http://host.docker.internal:8003 --admin-insecure-mode --public-invites --auto-ping-connection --auto-accept-invites --auto-accept-requests --auto-respond-credential-proposal --auto-respond-credential-request --auto-respond-presentation-proposal --auto-verify-presentation --genesis-url http://host.docker.internal:9000/genesis --seed 00000000000000000000000000000007 --auto-provision --recreate-wallet --wallet-type indy --wallet-name issuer --wallet-key mykey

From Bob, /issue-credential-2.0/send-request with the following credential
```
{
    "connection_id": "....",
    "filter": {
        "ld_proof": {
            "credential": {
                "@context": [
                    "https://www.w3.org/2018/credentials/v1",
                    "https://w3id.org/citizenship/v1",
                    "https://w3id.org/security/bbs/v1"
                ],
                "type": [
                    "VerifiableCredential",
                    "PermanentResidentCard"
                ],
                "issuer": "....",
                "identifier": "83627465",
                "name": "Permanent Resident Card",
                "description": "Government of Example Permanent Resident Card 1.",
                "issuanceDate": "2010-01-01T19:53:24Z",
                "expirationDate": "2029-12-03T12:19:52Z",
                "credentialSubject": {
                    "id": "....",
                    "type": [
                        "PermanentResident",
                        "Person"
                    ],
                    "givenName": "JOHN",
                    "familyName": "SMITH",
                    "gender": "Male",
                    "image": "data:image/png;base64,iVBORw0KGgokJggg==",
                    "residentSince": "2015-01-01",
                    "lprCategory": "C09",
                    "lprNumber": "999-999-999",
                    "commuterClassification": "C1",
                    "birthCountry": "Bahamas",
                    "birthDate": "1958-07-17"
                }
            },
            "options": {
                "proofType": "Ed25519Signature2018"
            }
        }
    },
    "holder_did": "...."
}
```

Alice receives the request and does not issue automatically as intended

**Comment by shaangill025** [comment]
#1667 fixes 3b.

**Comment by shaangill025** [comment]
Regarding 2. I am not able to replicate the problem and the `cred_rev_id [from IssuerCredRevRecord]` is correct in the thread_id for both v1 and v2.

File: aries_cloudagent/tails/indy_tails_server.py [hunk 1]
```diff
@@ -30,9 +32,21 @@ async def upload_tails_file(
             backoff: exponential backoff in retry interval
             max_attempts: maximum number of attempts to make
         """
-
-        genesis_transactions = context.settings.get("ledger.genesis_transactions")
         tails_server_upload_url = context.settings.get("tails_server_upload_url")
+        if not (context.settings.get("ledger.genesis_transactions")):
+            if isinstance(context, Profile):
```
Comment (TimoGlastra) [inline aries_cloudagent/tails/indy_tails_server.py:37]: Maybe we can update the calls to the `upload_tails_file` method to always pass the context? That way we don't have to do an instance check on the type

File: aries_cloudagent/tails/indy_tails_server.py [hunk 2]
```diff
@@ -30,9 +32,21 @@ async def upload_tails_file(
...
+                await context.injector.inject(
+                    BaseMultipleLedgerManager
+                ).get_write_ledger()
+            )[1].pool
+            try:
+                genesis_transactions = pool.genesis_transactions
+            except AttributeError:
+                genesis_transactions = pool.genesis_txns_cache
+        else:
+            genesis_transactions = context.settings.get("ledger.genesis_transactions")
```
Comment (TimoGlastra) [inline aries_cloudagent/tails/indy_tails_server.py:49]: I think we could simplify this a bit and split up getting the pool into multiple lines for readability:

```suggestion
        genesis_transactions = context.settings.get("ledger.genesis_transactions")
        
        if not genesis_transactions:
            ledger_manager = context.injector.inject(BaseMultipleLedgerManager)    
            write_ledgers = await ledger.manager.get_write_ledger()
            pool = write_ledgers[0].pool
        
            try:
                genesis_transactions = pool.genesis_transactions
            except AttributeError:
                genesis_transactions = pool.genesis_txns_cache
```

File: requirements.bbs.txt [hunk 3]
```diff
@@ -1 +1 @@
-ursa-bbs-signatures==1.0.1
\ No newline at end of file
+ursa-bbs-signatures~=1.0.1
```
Comment (TimoGlastra) [inline requirements.bbs.txt:1]: Do we want to make this change as part of this PR?

File: aries_cloudagent/tails/indy_tails_server.py (hunk 1 above)
Comment (shaangill025) [reply aries_cloudagent/tails/indy_tails_server.py]: Updated accordingly

File: aries_cloudagent/tails/indy_tails_server.py (hunk 2 above)
Comment (shaangill025) [reply aries_cloudagent/tails/indy_tails_server.py]: Updated accordingly

**Pull request closed without merging**